core/src/hlist.rs

```diff
@@ -1546,6 +1569,48 @@ where
     }
 }
 
+/// Trait for extending the contents of an HList with another HList
+///
+/// This is the trait part of the implementation of the inherent method
+/// [`HCons::extend`]. Please see that method for more information.
+///
+/// [`HCons::extend`]: struct.HCons.html#method.extend
+pub trait HExtend<T: HList> {
+    /// The result of the combination
+    type Output: HList;
+
+    /// Extends `self` with `other`
+    fn extend(self, other: T) -> Self::Output;
+}
+
+impl<T: HList> HExtend<T> for HNil {
```

Comment:
Nice, can you please add a test case for HNil extend HNil and Hcons extend HNil cases?

Reply:
Done, with bonus HNil extend HCons case too.